reya-rust-sdk/src/http_provider.rs

```diff
@@ -606,6 +633,150 @@ fn decode_reason(reason_bytes: Bytes) -> (String, ReasonError) {
+            }
+            RpcErrorsErrors::CollateralPoolCollision(collateral_pool_collision) => {
+                error!("reason error={:?}", collateral_pool_collision);
+                return (
+                    String::from("CollateralPoolCollision"),
+                    AEReasonError::CollateralPoolCollision,
+                );
+            }
+            RpcErrorsErrors::AccountBelowIM(account_below_im) => {
+                error!("reason error={:?}", account_below_im);
+                return (
+                    String::from("AccountBelowIM"),
+                    AEReasonError::AccountBelowIM,
+                );
+            }
+            RpcErrorsErrors::CollateralIsNotQuote(collateral_is_not_quote) => {
+                error!("reason error={:?}", collateral_is_not_quote);
+                return (
+                    String::from("CollateralIsNotQuote"),
+                    AEReasonError::CollateralIsNotQuote,
+                );
+            }
+            RpcErrorsErrors::CollateralPoolNotFound(collateral_pool_not_found) => {
+                error!("reason error={:?}", collateral_pool_not_found);
+                return (
+                    String::from("CollateralPoolNotFound"),
+                    AEReasonError::CollateralPoolNotFound,
+                );
+            }
+            RpcErrorsErrors::WithinBubbleCoverageNotExhausted(within_bubble) => {
+                error!("reason error={:?}", within_bubble);
+                return (
+                    String::from("WithinBubbleCoverageNotExhausted"),
+                    AEReasonError::WithinBubbleCoverageNotExhausted,
+                );
+            }
+            RpcErrorsErrors::AccountNotEligibleForAutoExchange(account_not_eligible) => {
+                error!("reason error={:?}", account_not_eligible);
+                return (
+                    String::from("AccountNotEligibleForAutoExchange"),
+                    AEReasonError::AccountNotEligibleForAutoExchange,
+                );
+            }
+            RpcErrorsErrors::CollateralCapExceeded(collateral_cap_exceeded) => {
+                error!("reason error={:?}", collateral_cap_exceeded);
+                return (
+                    String::from("CollateralCapExceeded"),
+                    AEReasonError::CollateralCapExceeded,
+                );
+            }
+            // all other errors are mapped to UnknownError
+            _ => {
+                info!("RPC error:{:?}", decoded_error);
+                return (
+                    format!("RPC error={:?}", decoded_error),
+                    AEReasonError::UnknownError,
+                );
+            }
+        },
+        Err(err) => {
+            return (format!("Error={:?}", err), AEReasonError::DecodingError);
+        }
+    }
+}
+
+fn handle_auto_exchange_error(e: Error) {
+    match e {
+        Error::AbiError(revert_reason) => {
+            info!("[Error decoding] ABI error: {:?}", revert_reason);
```

Comment:
errors should be logged as error! in all cases

Reply:
true, changed all info from this fn into error in [2d88303](https://github.com/Reya-Labs/reya-rust-sdks-monorepo/pull/24/commits/2d883034e2956bb6b0fcdd9eace3cbab6af9b46c)